lib/html_proofer/cache.rb

```diff
@@ -93,8 +93,10 @@ def retrieve_urls(urls_detected, type)
       # if there are no urls, bail
       return {} if urls_detected.empty?
 
-      urls_detected = urls_detected.transform_keys do |url|
-        cleaned_url(url)
+      if type == :external
+        urls_detected = urls_detected.transform_keys do |url|
+          cleaned_url(url)
+        end
       end
```

Comment:
Actually, I think we may want to only avoid `escape_unescape` for the external links, since the other part of `cleaned_url` that handles trailing "/", "#", "?" would still be relevant for internal links. Any input on this @gjtorikian?
If we keep part of `cleaned_url` for internal links, we should probably also include it to `add_internal` for consistency, similar to what is done for `add_external`